hibernate-core/src/main/java/org/hibernate/boot/model/source/internal/hbm/ModelBinder.java

```diff
@@ -3494,7 +3504,16 @@ public Identifier determineImplicitName(final LocalMetadataBuildingContext conte
 
 				getCollectionBinding().setElement( elementBinding );
 
-				getCollectionBinding().setManyToManyWhere( elementSource.getWhere() );
+				final PersistentClass referencedEntityBinding = mappingDocument.getMetadataCollector().getEntityBinding(
+						elementSource.getReferencedEntityName()
+				);
+				getCollectionBinding().setWhere( getPluralAttributeSource().getWhere() );
```

Comment:
When is this one used and why is it different?

Reply:
I've added comments to clarify.